libs/server/Resp/RespCommandInfoParser.cs

```diff
@@ -0,0 +1,439 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Text;
+using Garnet.common;
+
+namespace Garnet.server
+{
+    /// <summary>
+    /// Logic for parsing command info from RESP format
+    /// </summary>
+    public class RespCommandInfoParser
+    {
+        /// <summary>
+        /// Tries to parse a RespCommandInfo object from RESP format
+        /// </summary>
+        /// <param name="ptr">Pointer to current RESP chunk to read</param>
+        /// <param name="end">Pointer to end of RESP chunk to read</param>
+        /// <param name="supportedCommands">Mapping between command name and Garnet RespCommand and ArrayCommand values</param>
+        /// <param name="commandInfo">Parsed RespCommandsInfo object</param>
+        /// <param name="parentCommand">Name of parent command, null if none</param>
+        /// <returns>True if parsing successful</returns>
+        public static unsafe bool TryReadFromResp(ref byte* ptr, byte* end, IReadOnlyDictionary<string, (RespCommand, byte?)> supportedCommands, out RespCommandsInfo commandInfo, string parentCommand = null)
+        {
+            commandInfo = default;
+
+            // Command info is null
+            if (Encoding.ASCII.GetString(ptr, 5) == "$-1\r\n") return true;
```

Comment:
We can avoid allocation by
```suggestion
            if (new ReadOnlySpan<byte>(ptr, 5).SequenceEquals("$-1\r\n"u8)) return true;
```
(Probably requires adding `using System;`, <sub>`<ImplicitUsings>` would be nice</sub>)